new-packages/ts-project/src/state.ts

```diff
@@ -348,6 +361,43 @@ function extractEdgeGroup(
               registerActionBlocks(ctx, blocks, edge.data.actions);
               return;
             }
+            case 'internal': {
+              if (seenInternalProp) {
+                // `reenter` was already seen
+                ctx.errors.push({
+                  type: 'property_mixed',
+                });
+                return;
+              }
+              seenInternalProp = true;
+
+              if (findProperty(ctx, ts, element, 'reenter')) {
+                return;
+              }
```

Comment:
Shouldn't we raise a property_mixed error in this case too? I'm confused about the difference between this line of code and the one above that checks for `seenInternalProp`.

Reply:
we'll raise it when we reach `reenter` - not raising it here avoids it being raised twice, IIRC. I agree that this is slightly weirdly written but remember that each of those can be seen first - so there is a little bit of a dance happening here to avoid reporting it twice and stuff.